rust-runtime/aws-smithy-types/src/timeout/config.rs

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+use std::fmt::{Debug, Formatter};
+
+/// Top-level configuration for timeouts
+///
+/// # Example
+///
+/// ```rust
+/// # use std::time::Duration;
+///
+/// # fn main() {
+/// use aws_smithy_types::{timeout, tristate::TriState};
+///
+/// let api_timeouts = timeout::Api::new()
+///     .with_call_timeout(TriState::Set(Duration::from_secs(2)))
+///     .with_call_attempt_timeout(TriState::Set(Duration::from_secs_f32(0.5)));
+/// let timeout_config = timeout::Config::new()
+///     .with_api_timeouts(api_timeouts);
+///
+/// assert_eq!(
+///     timeout_config.api.call_timeout(),
+///     TriState::Set(Duration::from_secs(2))
+/// );
+///
+/// assert_eq!(
+///     timeout_config.api.call_attempt_timeout(),
+///     TriState::Set(Duration::from_secs_f32(0.5))
+/// );
+/// # }
+/// ```
+#[derive(Clone, PartialEq, Default)]
+pub struct Config {
+    /// API timeouts used by Smithy `Client`s
+    pub api: super::Api,
+    /// HTTP timeouts used by `DynConnector`s
+    pub http: super::Http,
+    /// TCP timeouts used by lower-level `DynConnector`s
+    pub tcp: super::Tcp,
+}
+
+impl Debug for Config {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            r#"Timeouts:
+{:?}
+{:?}
+{:?}
+"#,
+            self.api, self.http, self.tcp,
+        )
```

Comment:
What's the reason for manually implementing Debug for this struct? Seems like it would be better to use the built-in one so that customers can decide if they want to print it all on one line or pretty print it.

Reply:
originally it was so I could write `(unset)` instead of `None` but I stopped doing that so there's no reason to keep this now